ui-src/src/EditProfile.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+
+class EditProfile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newNameText: ""
+    };
+  }
+  componentDidMount() {
+    const { firstName } = this.props;
+    this.setState({ newNameText: firstName });
+  }
+  updateNameText = e => {
+    this.setState({
+      newNameText: e.target.value
+    });
+  };
+  onHandleSubmit = e => {
+    const { newNameText } = this.state;
+    const { setFirstName } = this.props;
+    e.preventDefault();
+
+    if (!newNameText) return;
+    setFirstName(newNameText);
```

Comment:
Once the action/request completes, the user expects some feedback. 
This could be done with some kind of transient user feedback message like 'Name was saved' that comes into the page, then out. Could also be done with a redirect away from the profile page. 

I will review how to go about treating an action as a Promise so that you can perform some code once the action completes.

Reply:
done - goes back to meow container. also added first name set to handle as a default.